internal/extsvc/types_test.go

```diff
@@ -211,3 +211,87 @@ func TestDecodeURN(t *testing.T) {
 		})
 	}
 }
+
+func TestUniqueCodeHostIdentifier(t *testing.T) {
+	for _, tc := range []struct {
+		config string
+		kind   string
+		want   string
+	}{
+		{
+			kind:   KindGitLab,
+			config: `{"url": "https://example.com"}`,
+			want:   "https://example.com/",
+		},
+		{
+			kind:   KindGitHub,
+			config: `{"url": "https://github.com"}`,
+			want:   "https://github.com/",
+		},
+		{
+			kind:   KindGitHub,
+			config: `{"url": "https://github.example.com/"}`,
+			want:   "https://github.example.com/",
+		},
+		{
+			kind: KindAWSCodeCommit,
+			config: `{
+				"region": "eu-west-1",
+				"accessKeyID": "accesskey",
+				"secretAccessKey": "secretaccesskey",
+				"gitCredentials": {
+					"username": "my-user",
+					"password": "my-password"
+				}
+			}`,
+			want: "eu-west-1:accesskey",
+		},
+		{
+			kind:   KindBitbucketServer,
+			config: `{"url": "https://bitbucket.sgdev.org/"}`,
+			want:   "https://bitbucket.sgdev.org/",
+		},
+
+		{
+			kind:   KindBitbucketCloud,
+			config: `{"url": "https://bitbucket.org/"}`,
+			want:   "https://bitbucket.org/",
+		},
+
+		{
+			kind:   KindGitolite,
+			config: `{"host": "ssh://git@gitolite.example.com:2222/"}`,
+			want:   "ssh://git@gitolite.example.com:2222/",
+		},
+		{
+			kind:   KindGitolite,
+			config: `{"host": "git@gitolite.example.com"}`,
+			want:   "git@gitolite.example.com/",
+		},
+		{
+			kind:   KindPerforce,
+			config: `{"p4port": "ssl:111.222.333.444:1666"}`,
```

Comment:
```suggestion
			config: `{"p4.port": "ssl:111.222.333.444:1666"}`,
```

Answer to my comment below.

Reply:
Nice catch!